apis/contour/v1beta1/ingressroute.go

```diff
@@ -73,22 +84,27 @@ type Service struct {
 	// Port (defined as Integer) to proxy traffic to since a service can have multiple defined
 	Port int `json:"port"`
 	// Weight defines percentage of traffic to balance traffic
+	// +optional
 	Weight uint32 `json:"weight,omitempty"`
 	// HealthCheck defines optional healthchecks on the upstream service
+	// +optional
 	HealthCheck *HealthCheck `json:"healthCheck,omitempty"`
 	// LB Algorithm to apply (see https://github.com/projectcontour/contour/blob/master/design/ingressroute-design.md#load-balancing)
+	// +optional
 	Strategy string `json:"strategy,omitempty"`
 	// UpstreamValidation defines how to verify the backend service's certificate
+	// +optional
 	UpstreamValidation *projcontour.UpstreamValidation `json:"validation,omitempty"`
 }
 
-// HTTPHealthCheckPolicy defines health checks on the upstream service.
+// HealthCheck defines health checks on the upstream service.
 type HealthCheck struct {
 	// HTTP endpoint used to perform health checks on upstream service
 	Path string `json:"path"`
 	// The value of the host header in the HTTP health check request.
 	// If left empty (default value), the name "contour-envoy-healthcheck"
 	// will be used.
+	// +optional
 	Host string `json:"host,omitempty"`
 	// The interval (seconds) between health checks
```

Comment:
I think this is optional

Reply:
I marked everything with 'omitempty' as optional, as I assumed it must be if you can omit it from the json.

Reply:
there are probably errors in our json encoding of those types as well.